### ISSUE TITLE
An --ema checkpoint is not a valid resume point: EMA weights are paired with raw optimizer state

### ISSUE THREAD

**Issue by lorenzoliuzzo** [lorenzoliuzzo/supervised-learning-on-food-images#44]
`src/main.py:575`:

```python
save_state = ema_model.module.state_dict() if ema_model is not None else model.state_dict()
```

The checkpoint stores the **smoothed** weights under `'state_dict'`, but `optimizer` and `scheduler` come from the **raw** trajectory. `--resume` (`src/main.py:491`) then loads the smoothed weights into `model` and pairs them with the raw run's momentum buffers.

Two things go wrong at once:
- the raw trajectory is silently discarded and replaced by its own moving average
- `ema_model` is reconstructed from scratch, so the averaging restarts with `n_averaged = 0` and the first `update_parameters` call re-seeds rather than blends

The checkpoint is correct as a **deliverable** -- the smoothed weights are the ones worth keeping, as the comment at `src/main.py:545` argues. It is not correct as a **resume point**.

### Fix

Store both, e.g. `state_dict` (raw, for resuming) plus `ema_state_dict` (smoothed, for evaluation and shipping), and have `--resume` restore the `AveragedModel` including its `n_averaged` buffer. Alternatively, refuse `--resume` on an `--ema` run and document why.

Note `benchmarks/analyze_errors.py:221` and `benchmarks/significance_test.py` both read `checkpoint['state_dict']` and expect the *evaluation* weights, so whichever key changes meaning has to be updated there too.
